### DISCUSSION TITLE
Use of tech in other applications than `cd`ing? Standalone CLI tool?

### DISCUSSION THREAD

**Discussion by silasb** [ajeetdsouza/zoxide#358] [General]
I've used `autocd` long ago (maybe 6 years ago) and recently started using `zoxide` and love it, but I'm now looking for new and interesting ways to do something similar with git branches.  Just like `fzf` created an awesome generic tool for doing fuzzy finding with an awesome UI I'm looking for something that is smarter and think `zoxide` tech could be a create tool for creating smart fuzzy finder.

**Comment by ajeetdsouza** [comment]
Hey, thanks! There's a post on using `git checkout` with fzf [here](https://polothy.github.io/post/2019-08-19-fzf-git-checkout/). Is that what you meant, or did you have something else in mind?

**Reply by silasb** [reply]
I've seen that and it's not what I want.  I'm looking to do `git checkout PLAT-490` and checkout `PLAT-490-bumping-ruby-to-xyz` or even having `git checkout 490` do the same thing since it's unique enough in my case.  Obviously I'd need to buildup the "DB" somehow and I'd likely do that through fetching origin or naturally, by manually checking out PR/branches as part of my workflow.

**Reply by ajeetdsouza** [reply]
That makes sense, but as far as I can tell, there's not much zoxide has in common with a program like this, other than structure. What you basically want is:
- Database serialization (zoxide uses [bincode](https://github.com/bincode-org/bincode))
- Atomic database writes (zoxide uses [tempfile](https://github.com/Stebalien/tempfile))
- Querying algorithm (zoxide's algorithm is specialized for paths, you'll have to roll your own)

Is there any functionality in particular that you'd like to expose?

**Reply by silasb** [reply]
Maybe this is my ignorance with zoxide and what it actually does behind the scenes.  I'll need to dive in over the weekend to better understand.  Thanks and I'll take a look into this.
